File: cpp/se/PollOperation.cpp
#include "se/PollOperation.hpp"

std::vector<const EnumType *> PollOperation::values;

const PollOperation* PollOperation::REQ()
{
	static const PollOperation po("req");
	return PollOperation::value("req");
}

const PollOperation* PollOperation::ACK()
{
	static const PollOperation po("ack");
	return PollOperation::value("ack");
}

PollOperation::PollOperation(const std::string &op)
    : EnumType(values, op)
{ }

const PollOperation* PollOperation::value (const std::string &name)
{
    return (const PollOperation *)EnumType::value (name, values);
}

void PollOperation::init()
{
    REQ();
    ACK();
}

File: src/main/java/com/ausregistry/jtoolkit2/se/PollOperation.java
package com.ausregistry.jtoolkit2.se;

/**
 * Enumeration of poll operations supported by EPP.
 */
public enum PollOperation {
    REQ("req"),
    ACK("ack");

    private String op;

    PollOperation(String opName) {
        op = opName;
    }

    public String toString() {
        return op;
    }
}

